homework-selects/sql_questions.sql

```diff
@@ -1,57 +1,579 @@
+            FROM employees
+        ) THEN 'LOW'
+        WHEN salary > (
+            SELECT AVG(salary) 
+            FROM employees
+        ) THEN 'HIGH'
+    ELSE NULL
+    END AS "SALARY STATUS"
+FROM employees;
+
 --34. all the employees who earn more than the average and who work in any of the IT departments.
+
+SELECT *
+FROM employees
+    JOIN departments USING (department_id)
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+    )
+    AND department_name LIKE '%IT%'
+	
 --35. who earns more than Mr. Ozer.
+
+SELECT *
+FROM employees
+    WHERE salary > (
+        SELECT salary
+        FROM employees
+            WHERE last_name || ' ' || first_name LIKE '%Ozer%';
+    )
+
 --36. which employees have a manager who works for a department based in the US.
+
+SELECT E.*
+FROM employees E
+    JOIN employees M
+        ON E.manager_id = M.employee_id
+    JOIN departments D
+        ON M.department_id = D.department_id
+    JOIN locations L
+        ON L.location_id = D.location_id
+    WHERE L.country_id = 'US';
+	
 --37. the names of all employees whose salary is greater than 50% of their department’s total salary bill.
+
+SELECT first_name || ' ' || last_name AS "FULL NAME"
+FROM employees
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+        WHERE department_id = department_id
+    );
+
 --38. the employee id, name ( first name and last name ), salary, department name and city for all
 --the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.  
+
+SELECT employee_id, 
+    first_name || ' ' || last_name AS "FULL NAME",
+    salary,
+    department_name,
+    city
+FROM employees
+    JOIN departments USING (department_id)
+    JOIN locations USING (location_id)
+    WHERE salary IN (
+        SELECT MAX(salary)
+        FROM employees
+            WHERE hire_date BETWEEN TO_DATE('01/01/2002', 'dd/mm/yyyy') AND
+                        TO_DATE('31/12/2003', 'dd/mm/yyyy')
+    )
+
 --39. the first and last name, salary, and department ID for all those employees who earn more than the average salary and arrange the list in descending order on salary.
+
+SELECT first_name, 
+    last_name,
+    salary,
+    department_id
+FROM employees
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+    )
+    ORDER BY salary DESC
+	
 --40. the first and last name, salary, and department ID for those employees who earn more than the maximum salary of a department which ID is 40.
+
+SELECT first_name,
+    last_name,
+    salary,
+    department_id
+FROM employees
+    WHERE salary > (
+        SELECT MAX(salary)
+        FROM employees
+        WHERE department_id = 40
+    );
+	
 --41. the department name and Id for all departments where they located, that Id is equal to the Id for the location where department number 30 is located.
+
+SELECT department_name, 
```

Comment:
you need to print the locationId not department id

Reply:
Fixed, but condition is strange

Reply:
at least there is description. sometimes there is posibility that you will recieve only task with name and no description, or description that you don;t understand fully. always contact the person who wrote it, or team members to inform that you don't get it properly